servers/src/test/java/tachyon/worker/block/TieredBlockStoreTest.java

```diff
@@ -297,4 +299,43 @@ public void freeSpaceWithBlockLockedTest() throws Exception {
     mBlockStore.freeSpace(USER_ID1, mTestDir1.getCapacityBytes(), mTestDir1.toBlockStoreLocation());
     Assert.assertEquals(mTestDir1.getCapacityBytes(), mTestDir1.getAvailableBytes());
   }
+
+  @Test
+  public void getBlockWriterForNonExistingBlockTest() throws Exception {
+    mThrown.expect(NotFoundException.class);
+    mThrown.expectMessage("Failed to get TempBlockMeta: temp blockId " + BLOCK_ID1 + " not found");
+
+    mBlockStore.getBlockWriter(USER_ID1, BLOCK_ID1);
+    Assert.assertTrue(mLockManager.getLockedBlocks().isEmpty());
```

Comment:
If `getBlockWriter` is expected to throw the exception, this assert will not be evaluated.